### PR TITLE
add register_device to register a single ios device more easily

### PR DESCRIPTION
**Pull request by pvinis**
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:

- [X] Run `bundle exec rspec` from the root directory.
- [X] Run `bundle exec rubocop -a` to ensure the code style is valid
- [X] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:
Issue: #7797

### CODE REVIEW

**Comment by KrauseFx** [comment]
Could you also make the `register_devices` action use the `register_device` action? This way we have far less duplicate code. Also make sure to add the new action to `unused_options_spec` to make the tests pass 👍

**Comment by KrauseFx** [comment]
Ah, just read your comment on https://github.com/fastlane/fastlane/issues/7797#issuecomment-271558296. 

> i made a pr but i didnt make `register_devices` call `register_device`, fistly because i dont know, and secondly because then it has to check all the time for credentials, and i dont know if that will be too slow.. what do you think?

Yeah, great point, it would login for every device, which is not something we want to do. Let's keep it like this for now then 👍

File: fastlane/lib/fastlane/actions/register_device.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        udid = params[:udid]
+
+        credentials = CredentialsManager::AccountManager.new(user: params[:username])
+        Spaceship.login(credentials.user, credentials.password)
+        Spaceship.select_team
+
+        UI.user_error!("Passed invalid UDID: #{udid} for device: #{name}") unless UDID_REGEXP =~ udid
+        Spaceship::Device.create!(name: name, udid: udid)
+
+        UI.success("Successfully registered new device.")
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/register_device.rb]: Let's remove the `.` from here

File: fastlane/lib/fastlane/actions/register_device.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                                       env_name: "DELIVER_USER",
+                                       description: "Optional: Your Apple ID",
+                                       default_value: user)
+        ]
+      end
+
+      def self.details
+        [
+          "This will register an iOS device with the Developer Portal so that you can include it in your provisioning profiles.",
+          "This is an optimistic action, in that it will only ever add a device to the member center, and never remove a device. If the device has already been registered within the member center, it will be left alone in the member center.",
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/register_device.rb]: > and never remove a device
I don't think that makes sense here

File: fastlane/lib/fastlane/actions/register_device.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          "The action will connect to the Apple Developer Portal using the username you specified in your `Appfile` with `apple_id`, but you can override it using the `username` option, or by setting the env variable `ENV['DELIVER_USER']`."
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/register_device.rb]: Let's remove mentioning the `DELIVER_USER` option here

File: fastlane/lib/fastlane/actions/register_device.rb [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+      def self.author
+        "pvinis"
+      end
+
+      def self.example_code
+        [
+          'register_device(
+            name: "Luka iPhone 6"
+            udid: "1234567890123456789012345678901234567890",
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/register_device.rb]: Please remove trailing `,`

**Comment by KrauseFx** [comment]
I can't believe yet another _fastlane_ contributor is from Vienna. You should come to this Thursday's CocoaHeads meeting in Vienna: https://www.facebook.com/events/698465373653477/

**Comment by pvinis** [comment]
I will fix those in a few minutes. hehe I was living in Vienna. I moved to Amsterdam in November.

**Review by pvinis — COMMENTED** [review]
i think that should be good

**Review by KrauseFx — APPROVED** [review]

**Comment by KrauseFx** [comment]
Looks good to me, thanks for your contribution. This will be in the next release 🚀

**Pull request merged**